### PR TITLE
Upgraded to new GCM API

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-9786

### CODE REVIEW

File: SwrveGoogleSDK/src/main/java/com/swrve/sdk/Swrve.java [hunk 1]
```diff
@@ -130,41 +130,16 @@ private void setRegistrationId(String regId) {
...
+                    }
                 } catch (Exception ex) {
                     Log.e(LOG_TAG, "Couldn't obtain the GCM registration id for the device", ex);
                 }
-
-                // Workaround Google Play bug
-                if (SwrveHelper.isNullOrEmpty(gcmRegistrationId)) {
-                    try {
-                        // Start pulling from broadcast listener
-                        String workaroundGcmRegId = null;
```
Comment (dominicmarmionswrve) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/Swrve.java]: Nice to get rid of this!

File: SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmBroadcastReceiver.java [hunk 2]
```diff
@@ -1,50 +0,0 @@
...
-import android.content.Intent;
-import android.support.v4.content.WakefulBroadcastReceiver;
-
-import com.swrve.sdk.SwrveHelper;
-
-/**
- * Used internally to receive push notifications inside for your app.
- */
-public class SwrveGcmBroadcastReceiver extends WakefulBroadcastReceiver {
-    private static String workaroundRegistrationId;
```
Comment (dominicmarmionswrve) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmBroadcastReceiver.java]: :dancer:

File: SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java [hunk 3]
```diff
@@ -30,14 +22,9 @@ public void onCreate() {
...
 
     @Override
-    protected void onHandleIntent(Intent intent) {
-        try {
-            handler.onHandleIntent(intent, GoogleCloudMessaging.getInstance(this));
-        } finally {
-            SwrveGcmBroadcastReceiver.completeWakefulIntent(intent); // Always release the wake lock provided by the WakefulBroadcastReceiver.
-        }
+    public void onMessageReceived(String from, Bundle data) {
+        processNotification(data);
```
Comment (dominicmarmionswrve) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java]: Where is processNotification method? 
Should this be handler.processNotification?

Comment (dominicmarmionswrve) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java]: Oh, I see it down below......

File: SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java [hunk 4]
```diff
@@ -8,19 +7,12 @@
...
 /**
  * Used internally to process push notifications inside for your app.
  */
-public class SwrveGcmIntentService extends IntentService implements ISwrveGcmService {
-
-    public SwrveGcmIntentService() {
-        super("SwrveGcmIntentService");
-    }
-
+public class SwrveGcmIntentService extends GcmListenerService implements ISwrveGcmService {
```
Comment (dominicmarmionswrve) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java]: I added the ISwrveGcmService to make it more testable, but I don't like it anymore. Think we should remove it.

File: SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java (hunk 3 above)
Comment (Sergio-Mira) [inline SwrveGoogleSDK/src/main/java/com/swrve/sdk/gcm/SwrveGcmIntentService.java]: I am keeping it to not change the API that some customers might be overriding

**Comment by dominicmarmionswrve** [comment]
A few debug logs might be handy as this can be tricky to debug when deployed.

**Pull request merged**